### ISSUE TITLE
Fit titleIconView to page width

### ISSUE THREAD

**Issue by fulldecent** [ealeksandrov/EAIntroView#212]
I was expecting the image to take up the available space. This is the code we used:

```swift
        let page1 = EAIntroPage()
        page1.title = NSLocalizedString("A party game", comment: "Intro screen 1 title")
        page1.desc = NSLocalizedString("For 3 to 12 players", comment: "Intro screen 1 detail")
        page1.titleIconView = UIImageView(image: UIImage(named: "help1"))
```

This is what we got

![screen shot 2017-10-03 at 10 44 50 pm](https://user-images.githubusercontent.com/382183/31157982-8fda5b76-a88c-11e7-8820-5252ed980bac.png)

Currently best practice is this implementation: https://github.com/kolyvan/kxintro

**Comment by ealeksandrov** [comment]
Have you tried using `bgImage` instead of `titleIconView`?
Also I have no idea what is happening on your attached screen and how do you want it to look.

**Comment by fulldecent** [comment]
I have tried bgImage too and got the same effect.

I was hoping for it to shrink to fit on the screen like in this screenshot.

![screenshot01](https://user-images.githubusercontent.com/382183/31194696-fcd4671a-a915-11e7-8517-2b1a8c55ba02.png)

**Comment by ealeksandrov** [comment]
What happens if you drop your image in example project?
I don't understand yet where is `titleIcon` on your screenshot.

**Comment by ealeksandrov** [comment]
Can you share this image with me, so I can test it?

**Comment by fulldecent** [comment]
You bet. Thanks for your help!

![help1](https://user-images.githubusercontent.com/382183/31282944-0d8d8c4a-aa82-11e7-878a-8dcfe7e0c49d.png)

**Comment by ealeksandrov** [comment]
![simulator screen shot - iphone 8 - 2017-10-06 at 19 01 34](https://user-images.githubusercontent.com/3132438/31287092-d2cad332-aac8-11e7-8f54-7e6dba454614.png)

Here is how you can do it now:
```obj-c
EAIntroPage *page1 = [EAIntroPage page];
page1.title = @"Testing image";
page1.desc = sampleDescription1;
page1.bgImage = [UIImage imageNamed:@"bg1"];

UIImage *testImage = [UIImage imageNamed:@"image"];
UIImageView *testImageView = [[UIImageView alloc] initWithImage:testImage];
float imageRatio = testImage.size.width / testImage.size.height;
float widthToFit = self.view.bounds.size.width;
[testImageView setFrame:CGRectMake(0, 0, widthToFit, widthToFit / imageRatio)];
page1.titleIconView = testImageView;
```

In EAIntroView we're preserving original image size 1:1. Code above will resize `titleIconView` to fit screen.
I will think more about updating page layout to handle this automatically.

**Comment by fulldecent** [comment]
Very nice, thank you!

In portrait orientation this will work. I'll need to figure it out when rotating or switching to other devices.

**Comment by ealeksandrov** [comment]
I'll look into it on weekend and give you update.

**Comment by ealeksandrov** [comment]
Added autolayout constraints to fit titleIconView inside page frame - https://github.com/ealeksandrov/EAIntroView/commit/a090a152d01e8259391432de867ba5c7b4aec7c7

With your image produces same screenshot as workaround above. Also works in landscape.
It lays out `titleIconView - title - description` vertically. PageControl and SkipButton may overlap - but they can be on any position on a page (for example, on top), so I don't manage them in this equation.

To add padding between description and screen bottom and fit pageControl there - you can add empty newlines at the end of page description.

Released in 2.11.0.

**Comment by fulldecent** [comment]
Thank you, this works great now!
